src/lib/react-query.tsx

```diff
@@ -14,6 +14,44 @@ import {isNative} from '#/platform/detection'
 export const labelersDetailedInfoQueryKeyRoot = 'labelers-detailed-info'
 const STORED_CACHE_QUERY_KEY_ROOTS = [labelersDetailedInfoQueryKeyRoot]
 
+async function checkIsOnline(): Promise<boolean> {
+  try {
+    const controller = new AbortController()
+    setTimeout(() => {
+      controller.abort()
+    }, 15e3)
+    const res = await fetch('https://public.api.bsky.app/xrpc/_health', {
+      cache: 'no-store',
+      signal: controller.signal,
+    })
+    const json = await res.json()
+    if (json.version) {
+      return true
+    } else {
+      return false
+    }
+  } catch (e) {
+    return false
+  }
+}
+
+let checkPromise: Promise<void> | undefined
+function checkIsOnlineIfNeeded() {
+  if (checkPromise) {
+    return
+  }
+  checkPromise = checkIsOnline().then(isOnline => {
+    checkPromise = undefined
+    onlineManager.setOnline(isOnline)
+  })
+}
+
+setInterval(() => {
+  if (AppState.currentState === 'active') {
+    checkIsOnlineIfNeeded()
+  }
+}, 5000)
```

Comment:
Wait so you're doing this constantly?

Reply:
Is there any way we could just trigger this when a request fails in a certain way?

Reply:
Just looked at upstream. I think if you started these checks once `network-error` gets fired and stopped them once it resolved, that should be effective.

I'm going to approve so that we can test on TF but we definitely should _not_ be pinging any of our infra once every 5s for the entirely uptime of the app

Reply:
The `network-error` event is unfortunately not enough. I am only seeing it being raised when the error happens during a session refresh. I’m not seeing it fire on regular failures so I don’t have a good way to observe them (unless we’re going to dig further into the change on the other side).

One possible alternative is to fork a fetch wrapper on the web too (currently we only do it on native) and add a handler there. I was hoping to avoid touching that part because @haileyok’s PR moves that around and at that point I think it would become more convenient. Maybe we should wait for that, or it’s worth forking it now?

Reply:
The other thing we could look at is the RQ onError callback but that wouldn’t cover everything, and I also couldn’t get it to work when I tried it last time.

Reply:
Yeah unfortunately we really can't move forward with it as-is.